.github/workflows/build-and-release.yml

```diff
@@ -28,13 +34,59 @@ jobs:
       version: ${{ steps.export.outputs.version }}
       runner-size: ${{ steps.export.outputs.runner-size }}
       infra-config-repo: ${{ steps.export.outputs.infra-config-repo }}
-
+      environment: ${{ steps.env-var.outputs.environment }}
+      commit-hash: ${{ steps.env-var.outputs.commit-hash }}
+      author-name: ${{ steps.env-var.outputs.author-name }}
+      core-repo-reference: ${{ steps.env-var.outputs.core-repo-reference }}
+      config-repo-reference: ${{ steps.env-var.outputs.config-repo-reference }}
 
     steps:
+      - id: env-var
```

Comment:
It looks to me like we might potentially be able to reduce this env-vars declaration through the use of the GitHub actions language. There's a lot going on here and on initial viewing I found it quite hard to dissect what the expected behaviour is. I'm wondering if it's worth us using the more-native GH actions syntax to do this if we can? It's hard to explain what I mean over GH comments, but do you mind if I have a go at a quick refactor of this on a branch of your branch to show you an example?